pixeltable/type_system.py

```diff
@@ -738,15 +739,24 @@ def to_tf(self) -> Union['tf.TypeSpec', Dict[str, 'tf.TypeSpec']]:
         return tf.TensorSpec(shape=(self.height, self.width, self.num_channels), dtype=tf.uint8)
 
     def validate_literal(self, val: Any) -> None:
-        # make sure file path points to a valid image file
-        if not isinstance(val, str):
-            raise TypeError(f'Expected file path, got {val}')
-        try:
-            _ = Image.open(val)
-        except FileNotFoundError:
-            raise TypeError(f'File not found: {val}')
-        except PIL.UnidentifiedImageError:
-            raise TypeError(f'File is not a valid image: {val}')
+        # make sure file path points to a valid image file or binary is a valid image
+        if not isinstance(val, (str, bytes)):
+            raise TypeError(f'Expected file path or binary, got {val}')
+
+        if isinstance(val, bytes):
+            try:
+                _ = Image.open(io.BytesIO(val))
+            except PIL.UnidentifiedImageError:
+                raise TypeError(f'Binary is not a valid image: {val}')
```

Comment:
By 'create test case' I meant: add some "with pytest.raises(...):" logic to the existing test_create_images().

Reply:
I ended up adding a new test for bad images, as it made sense to test both inline and file. let me know if that works. 

This is out of scope for this review, but, as an aside, erroring an insert upon finding a bad image may be undesirable from user perspective, at least from a bulk load situation. There always are bad images in a large collection, and at least for most datasets this is not a problem, as no specific image is crucial, and the user really means 'load the ones that work'. 

It is definitely useful the general idea of preventing errors downstream produced by invalid images, which are also annoying as the whole job may fail due to a single image failing. 

The problem is finding bad images can itself be a slow job due to slow storage, and keeping this meta-information around for the user is helpful.

Reply:
Good point, and it sounds like it's worth thinking this through.

For computed columns, errors are explicitly recognized and recorded (as separate columns errortype and -msg, which  are exposed as properties of a column: eg, "computed_col_name.errortype"). They're also handled during expression evaluation (take a look at Evaluator.eval()). Something similar could be done for input data, ie, it just records the errors and skips all derived computed columns, but otherwise keeps going. This should probably be a table parameter, it may  not be appropriate in every situation.

But definitely worth adding if this makes it more attractive for researchers or for experimentation in general.